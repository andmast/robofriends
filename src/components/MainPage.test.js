import { shallow } from "enzyme";
import React from "react";
import MainPage from "./MainPage";

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

it("Expect to render Mainpage Componet", () => {
  expect(wrapper).toMatchSnapshot();
});

it("Returns the matching robots from matched search field", () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 3,
        name: "Andrea",
        email: "a@a.com",
      },
    ],
    searchField: "a",
    isPending: false,
  };
  const filteredRobots = mockProps2.robots;
  const wrapper2 = shallow(<MainPage {...mockProps2} />);
  expect(wrapper2.instance().filterRobots()).toEqual(filteredRobots);
});

it("Returns no matching robots from unmatched search field", () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 3,
        name: "Andrea",
        email: "a@a.com",
      },
    ],
    searchField: "z",
    isPending: false,
  };
  const filteredRobots = [];
  const wrapper2 = shallow(<MainPage {...mockProps2} />);
  expect(wrapper2.instance().filterRobots()).toEqual(filteredRobots);
});
