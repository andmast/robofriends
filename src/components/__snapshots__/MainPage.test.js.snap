// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Mainpage Componet 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <CardList
      robots={[Function]}
    />
  </Scroll>
</div>
`;
